File: SSVV1/src/main/java/Main.java
import console.*;
import domain.*;
import repository.*;
import service.*;
import validation.*;

public class Main {
    public static void main(String[] args) {
        Validator<Student> studentValidator = new OldStudentValidator();
        Validator<Assignment> temaValidator = new AssignmentValidator();
        Validator<Grade> notaValidator = new GradeValidator();

        StudentXmlRepository fileRepository1 = new StudentXmlRepository(studentValidator, "studenti.xml");
        AssignmentXmlRepository fileRepository2 = new AssignmentXmlRepository(temaValidator, "teme.xml");
        GradeXmlRepository fileRepository3 = new GradeXmlRepository(notaValidator, "note.xml");

        Service service = new Service(fileRepository1, fileRepository2, fileRepository3);
        UI consola = new UI(service);
        consola.run();

        //PENTRU GUI
        // de avut un check: daca profesorul introduce sau nu saptamana la timp
        // daca se introduce nota la timp, se preia saptamana din sistem
        // altfel, se introduce de la tastatura
    }
}

File: SSVV1/src/main/kotlin/Main.kt
import console.UI
import repository.GradeXmlRepository
import repository.StudentXmlRepository
import repository.AssignmentXmlRepository
import service.Service
import validation.GradeValidator
import validation.StudentValidator
import validation.AssignmentValidator

fun main() {
    val studentValidator = StudentValidator()
    val assignmentValidator = AssignmentValidator()
    val gradeValidator = GradeValidator()

    val studentRepository = StudentXmlRepository(studentValidator, "studenti.xml")
    val assignmentRepository = AssignmentXmlRepository(assignmentValidator, "teme.xml")
    val gradeRepository = GradeXmlRepository(gradeValidator, "note.xml")

    val service = Service(studentRepository, assignmentRepository, gradeRepository)
    val console = UI(service)
    console.run()
}